### ISSUE TITLE
Curious on your thoughts / comparison of TypeStat to ts-migrate? Room for both? One or the other?

### ISSUE THREAD

**Issue by GitMurf** [JoshuaKGoldberg/TypeStat#1317]
As the title says, I am curious on whether there is an opinion on whether TypeStat can and/or should be used together with (or instead of) ts-migrate? Do they do different things well? Or are they essentially the same type of tool?

Thanks for your amazing work as always!

**Comment by JoshuaKGoldberg** [comment]
Yeah that's a great question! https://github.com/airbnb/ts-migrate is a great package and works really well. I think that right now, because of how early stage & buggy (🙃) TypeStat is, I'd probably recommend teams use ts-migrate over TypeStat for now. 

The reason why TypeStat isn't as "production ready" as ts-migrate is that it's got much larger aims as a project. ts-migrate gets you to a starting place as conveniently and quickly as possible. It suppresses type errors but doesn't do much about getting you to the best possible types. TypeStat, on the other hand, aims to do that convenient+quick work _plus_ improving your types as much as possible. Improving types is a very difficult, even unsolvable challenge!

Medium-to-long term, I think TypeStat will be more "competitive" _(in a friendly, supportive sense!)_:

* #1314: I'm going to split this library into >=2 pieces: an _initializer_ and an _enhhancer_. The _initializer_ piece will be a much more direct equivalent to ts-migrate.
* #1000: Now that I have a much better, more thorough understanding of the problem space, I'd like to rearchitect the _enhancer_ piece to be more stable and less convoluted to develop in

[My plans for 2023](https://www.joshuakgoldberg.com/blog/my-plans-for-2023#typestat) didn't start off with #1000 or #1314. I think I'm going to end up making a lot of progress on #1314 this season in service of #1318.

---

All that is to say: yes, resolving this docs issue is part of #1314. Great question & thanks for asking!

**Comment by GitMurf** [comment]
Love it all! I don't have any bandwidth right now to help but as soon as I get some I will be happy to try and help if I can.

A couple notes, ts-migrate appears to have quite a few bugs today given it hasn't been updated in a bit and my hunch is a lot to do with TS 5.0 as well as older eslint (core part of its workflow). Things like the "satisfies" keyword is not handled. And another big issue I had was JSX comments as they are not properly `{/* @ts-expect ... *\}` in curly braces. See here: https://github.com/airbnb/ts-migrate/issues/150

Also would be nice to have a more generalized (non react specific) workflow as I use solid-js for example.

Additionally a real bummer is when you try to re-run ts-migrate it actually reverts a lot of manual fixes to the Jsx comments for example :( point is, there is definitely a need for what you are working on today more than ever! Even just the ts-initialize piece will be very beneficial with an updated look with ts 5+ 👍

**Comment by GitMurf** [comment]
Curious where things sit today? We are preparing to finish off the rest of our js to ts migration by running either ts-migrate or typestat. We are doing everything net new in TS but still have all our grandfathered js code that gives us all the fun red squigglies. Our main goal is to draw a line in the sand so that we can actually add tsc type checking to our CLI and know when new errors are introduced... which we cannot do with our 600 leftover errors. Thanks as always for the amazing work in the TS community!

**Comment by JoshuaKGoldberg** [comment]
Things haven't changed at all since you last checked in, I'm afraid. I've put this on my list of 2024 initiatives - so you can expect something within the next ~11 months or so.

...I imagine that's probably not _super_ helpful given you're finishing off the rest of your migration now, sorry. 😓

**Comment by GitMurf** [comment]
@JoshuaKGoldberg I want to be mindful of abusing gh issues with stuff that doesn't belong there. Would you be open to adding gh discussions to this repo to have a place where the community can ask questions and discuss amongst ourselves any tips, tricks, learnings, problems etc.? Thanks!

**Comment by JoshuaKGoldberg** [comment]
Sure, just opened it up - thanks for asking! ❤️

**Comment by GitMurf** [comment]
> Sure, just opened it up - thanks for asking! ❤️

Man you're fast! Thank you good sir!

**Comment by GitMurf** [comment]
@JoshuaKGoldberg actually maybe this issue itself is a good candidate to convert to a Discussion item? I don't think I am able to do it. Thoughts?
